[AIRFLOW-4559] JenkinsJobTriggerOperator bugfix

uncletoxa opened the pull request:
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4559

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Previous version provides url to Request object implicitly. It creates weird interaction assigning it to `method` parameter instead of `url` which raises `requests.exceptions.MissingSchema: Invalid URL 'None': No schema supplied`. Fixed making both explicit.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Uses existing Operator tests.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

uncletoxa committed:
[AIRFLOW-4559] JenkinsJobTriggerOperator bugfix

Previous version provides url to Request object implicitly. It
creates weird interaction assigning it to 'method' parameter
instead of 'url' which raises
'requests.exceptions.MissingSchema: Invalid URL 'None': No schema
supplied'. Fixed making both explicit.

ashb reviewed (commented):
Can you add some unit tests?

uncletoxa committed:
[AIRFLOW-4559] JenkinsJobTriggerOperator bugfix tests

Add tests to check JenkinsJobTriggerOperator's build_job method
constructs correct Request object (specifically testing params
'url' and 'method').

uncletoxa committed:
test location in headers
